modules/gke-cluster-autopilot/variables.tf

```diff
@@ -35,6 +35,16 @@ variable "backup_configs" {
   nullable = false
 }
 
+variable "control_plane_endpoints_config" {
+  description = "Configuration for all of the cluster's control plane endpoints."
+  type = object({
+    dns_endpoint_config = optional(object({
+      allow_external_traffic = optional(bool)
+    }))
+  })
+  default = null
+}
+
```

Comment:
I know this aligns with the API does, but I don't like the type of this variable. I'm also concerned that now we have endpoint configuration spread across multiple variables.

I really like what the UI is doing now:
![image](https://github.com/user-attachments/assets/3c9d0992-f1e8-4ed3-8429-9b174bd0d145)

Could we consider extending this variable to include all control plane endpoint configuration options (DNS, PSC, and PSA)?

Something like
```terraform
variable "control_plane_endpoints_config" {
  description = "Configuration for all of the cluster's control plane endpoints."
  type = object({
    dns_endpoint = optional(object({...}))
    psc_endpoint = optional(object({...}))
    private_endpoint = optional(object({...}))
  })
}
```

I realize this is probably a larger change than what you had in mind. What do you think?

Reply:
I think it would be fine, though it makes a bit wary to differ a lot from the underlying Terraform resource (though granted, it has become a bit messy over the years). But it could look something like this:

```hcl
variable "control_plane_endpoints_config" {
  description = "Configuration for all of the cluster's control plane endpoints."
  type = object({
    dns_endpoint = optional(object({
     allow_external_traffic = optional(bool) 
   }))
    psc_endpoint = optional(object({
      subnetwork = optional(string) # maps to vpc_config.master_endpoint_subnetwork
      # enabling PSC implies no master_ipv4_cidr_block, peering_configs etc
      master_global_access = optional(bool) #  maps to private_cluster_config.master_global_access, I think this is only for PSC?
   }))
    private_endpoint = optional(object({ # Maps to private_cluster_config.enable_private_endpoint 
     # put vpc_config.master_ipv4_cidr_block here too ?
     # and perhaps private_cluster_config.peering_config
   }))
   master_authorized_ranges = optional(map(string)) # maps to vpc_config.master_authorized_networks, top-level because it applies to public/private/PSC
  })
}
```

If we don't need backwards compatibility, it would be fairly straightforward I suppose.